src/common/copilot/dataverseMetadata.ts

```diff
@@ -76,35 +78,42 @@ export function getEntityName(telemetry: ITelemetry, sessionID:string, dataverse
             const activeFileName = path.basename(absoluteFilePath); //"Copilot-Student-Loan-Registration-56a4.basicform.custom_javascript.js"
             const fileNameFirstPart = activeFileName.split('.')[0]; //"Copilot-Student-Loan-Registration-56a4"
 
-            const matchingFiles = getMatchingFiles(activeFileFolderPath, fileNameFirstPart); // ["Copilot-Student-Loan-Registration-56a4.basicform.yml"]
+            const matchingFiles = await getMatchingFiles(activeFileFolderPath, fileNameFirstPart); // ["Copilot-Student-Loan-Registration-56a4.basicform.yml"]
 
-            if (matchingFiles[0]) {
+            if (IS_DESKTOP && matchingFiles[0]) {
+                const diskRead = await import('fs');
                 const yamlFilePath = path.join(activeFileFolderPath, matchingFiles[0]);
-                const yamlContent = fs.readFileSync(yamlFilePath, 'utf8');
+                const yamlContent = diskRead.readFileSync(yamlFilePath, 'utf8');
                 const parsedData = parseYamlContent(yamlContent, telemetry, sessionID, dataverseEntity);
                 entityName = parsedData['adx_entityname'] || parsedData['adx_targetentitylogicalname'];
+            } else if (!IS_DESKTOP) {
+                entityName = getFileLogicalEntityName(document.uri.fsPath);
             }
         }
     } catch (error) {
-        sendTelemetryEvent(telemetry, { eventName: CopilotGetEntityFailureEvent, copilotSessionId:sessionID, dataverseEntity: dataverseEntity, error: error as Error});
+        sendTelemetryEvent(telemetry, { eventName: CopilotGetEntityFailureEvent, copilotSessionId: sessionID, dataverseEntity: dataverseEntity, error: error as Error });
```

Comment:
you sending over the 'dataverseEntity'?
If it's just the name of the entity, it would be best to rename to 'dataverseEntityName' so it's clearer when reviewing telemetry that we aren't sending customer data.